bundle/src/main/java/com/cognifide/cq/cqsm/foundation/actions/define/Define.java

```diff
@@ -57,9 +60,13 @@ public boolean isGeneric() {
 	}
 
 	@Override
-	public Map<String, String> provideDefinitions() {
-		return new HashMap<String, String>() {{
-			put(name, value);
-		}};
+	public Map<String, String> provideDefinitions(Map<String, String> alreadyDefined) {
+		Map<String, String> result = Collections.emptyMap();
+		boolean definitionExists = alreadyDefined.containsKey(name);
+		boolean shouldBeAdded = !(definitionExists && ifNotExists);

```

Comment:
`ifNotExists` seems quite vague in this context. Maybe something like `ignoreIfExists`? `dontOverride'?